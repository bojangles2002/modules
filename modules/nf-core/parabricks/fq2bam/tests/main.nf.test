nextflow_process {

    name "Test Process PARABRICKS_FQ2BAM"
    script "../main.nf"
    process "PARABRICKS_FQ2BAM"

    tag "modules"
    tag "modules_nfcore"
    tag "parabricks"
    tag "parabricks/fq2bam"

    test("parabricks_fq2bam_pe_default") {

        when {
            process {
                """
                input[0] = [
                   [ id:'reads', single_end:false],
                   [
                       file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                       file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                   ],
                   []
               ]
               input[1] = [
                   [id: 'interval_file'],
                   file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
               ]
               input[2] = [[id: 'fasta'],]
               input[3] = [[id: 'index'],]
               input[4] = [[id: 'known_sites'],]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("parabricks_fq2bam_se_default") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true],
                    [file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true)],
                    []
                ]

                input[1] = [
                    [id: 'reference'],
                    file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
                ]
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("parabricks_fq2bam_se_default_stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true],
                    [file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true)],
                    []
                ]

                input[1] = [
                    [id: 'reference'],
                    file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
                ]
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
