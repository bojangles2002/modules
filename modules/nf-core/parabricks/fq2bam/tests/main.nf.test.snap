{
    "sarscov2 - fastq.gz - paired-end": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "bai": [
                    
                ],
                "bam": [
                    
                ],
                "bqsr_table": [
                    
                ],
                "duplicate_metrics": [
                    
                ],
                "qc_metrics": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-29T16:10:31.241406"
    },
    "sarscov2 - fastq.gz - single-end": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "bai": [
                    
                ],
                "bam": [
                    
                ],
                "bqsr_table": [
                    
                ],
                "duplicate_metrics": [
                    
                ],
                "qc_metrics": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-29T16:10:40.006796"
    }
}